soc/intel/intel_adsp/ace/include/adsp_memory_regions.h

```diff
@@ -6,22 +6,38 @@
 #ifndef ZEPHYR_SOC_INTEL_ADSP_MEMORY_REGIONS_H_
 #define ZEPHYR_SOC_INTEL_ADSP_MEMORY_REGIONS_H_
 
-/* Define amount of regions other than core heaps that virtual memory will be split to
- * currently includes shared heap and oma region and one regions set to 0 as for table
- * iterator end value.
- */
-#define VIRTUAL_REGION_COUNT 3
-
-#define CORE_HEAP_SIZE 0x100000
-#define SHARED_HEAP_SIZE 0x100000
-#define OPPORTUNISTIC_REGION_SIZE 0x100000
-
 /* size of TLB table */
 #define TLB_SIZE DT_REG_SIZE_BY_IDX(DT_INST(0, intel_adsp_mtl_tlb), 0)
 
 /* Attribiutes for memory regions */
+#if CONFIG_MM_DRV_INTEL_ADSP_TLB_USE_PER_CORE_VIRTUAL_MEMORY_REGIONS
+#define VIRTUAL_REGION_COUNT_PER_CORE CONFIG_MP_MAX_NUM_CPUS
 #define MEM_REG_ATTR_CORE_HEAP 1U
+#else
+#define VIRTUAL_REGION_COUNT_PER_CORE 0
```

Comment:
if selected, this option would break SOF - both with and without https://github.com/thesofproject/sof/pull/10044